hair_salon/data/ir_actions_server.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<odoo>
+    <record id="break_booking_server_action" model="ir.actions.server">
+        <field name="code"><![CDATA[# Fetch related records by field 'x_appointment_part_2'
+related_records_1 = env['calendar.event'].search([('x_linked_to', '=', record.id)])
+related_records_1.unlink()]]></field>
+        <field name="model_id" ref="calendar.model_calendar_event"/>
+        <field name="state">code</field>
+        <field name="name">Execute Code</field>
+    </record>
+    <record id="delete_subpart_main_calendar_event_server_action" model="ir.actions.server">
+        <field name="child_ids" eval="[(6, 0, [ref('break_booking_server_action')])]"/>
+        <field name="code"><![CDATA[parent_record = env['calendar.event'].search([('id', '=', record.x_linked_to.id)])
+if parent_record.x_appointment_part_2.id == record.id:
+        parent_record.x_appointment_part_2.unlink()
+else:
+        parent_record.x_appointment_part_2.unlink()
+parent_record["active"] = False]]></field>
```

Comment:
Probably safer to `unlink` (as there may be more stuff to do)

Reply:
when you set `record['active'] = False`, you unlink. But when you do `record.unlink()`, it also unlinks the record, but there may be other related stuff to do that the CRUD method would do and that you may forget using the first way.